### PR TITLE
AirfRANS: clean default baseline

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

AirfRANS has no meaningful production baseline on the radford branch — only 1-epoch shakedown runs exist. This experiment establishes the first real baseline: the default senpai_transolver config trained for 150 epochs on AirfRANS full task. All other AirfRANS experiments will be compared against this run.

## Instructions

```bash
cd target/icml2026
python train.py \
  --dataset airfrans \
  --airfrans-task full \
  --epochs 150 \
  --lr 3e-4 \
  --optimizer lion \
  --batch-size 2 \
  --model senpai_transolver \
  --model-layers 3 \
  --model-hidden-dim 192 \
  --model-heads 3 \
  --model-slices 96 \
  --model-mlp-ratio 4 \
  --model-dropout 0.0 \
  --cosine-t-max 150 \
  --use-ema \
  --ema-decay 0.999 \
  --ema-start-step 50 \
  --use-lookahead \
  --surface-refine \
  --surface-refine-hidden 128 \
  --surface-refine-layers 2 \
  --weight-decay 1e-4 \
  --wandb-name "violet/airfrans-baseline-default" \
  --wandb-group "radford-airfrans-round1"
```

## Baseline

No AirfRANS production baseline exists on radford yet — this IS the baseline run.

- **Primary metric:** `val_primary/surface_mse` (lower is better)
- **Also report:** `val_primary/volume_mse`, per-channel MSEs for `Ux`, `Uy`, `p`, `nut`
- AirfRANS official task: `full` split

## Results

_To be filled in by the student._

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT violet:

## Results

### Primary Metrics

| Metric | Best Val (Epoch 4) | Test |
|--------|-------------------|------|
| **surface_mse** | **0.3973** | **0.5319** |
| volume_mse | 0.8890 (ep3) | 0.9146 |

### Per-Channel Surface MSE (Test)

| Channel | MSE |
|---------|-----|
| Ux | 0.00141 |
| Uy | 0.00422 |
| p | 2.1194 |
| nut | 0.00263 |

### Per-Channel Volume MSE (Test)

| Channel | MSE |
|---------|-----|
| Ux | 0.8913 |
| Uy | 0.9821 |
| p | 0.8860 |
| nut | 0.8988 |

### Training Details

- **Epochs completed:** 6 / 150 requested (hit SENPAI_TIMEOUT_MINUTES=30 limit)
- **Best val_primary/surface_mse:** 0.3973 at epoch 4
- **Final training loss:** 1.309 (epoch 6)
- **Peak memory:** Not tracked in train.py (no `torch.cuda.max_memory_allocated` logging)
- **W&B run ID:** `ditvmxcl`
- **W&B link:** https://wandb.ai/wandb-applied-ai-team/senpai-v1/runs/ditvmxcl

### Command

```bash
cd target/icml2026 && CUDA_VISIBLE_DEVICES=0 python train.py \
  --dataset airfrans \
  --airfrans-task full \
  --epochs 150 \
  --lr 3e-4 \
  --optimizer lion \
  --batch-size 2 \
  --model senpai_transolver \
  --model-layers 3 \
  --model-hidden-dim 192 \
  --model-heads 3 \
  --model-slices 96 \
  --model-mlp-ratio 4 \
  --model-dropout 0.0 \
  --cosine-t-max 150 \
  --use-ema \
  --ema-decay 0.999 \
  --ema-start-step 50 \
  --use-lookahead \
  --surface-refine \
  --surface-refine-hidden 128 \
  --surface-refine-layers 2 \
  --weight-decay 1e-4 \
  --agent violet \
  --wandb-name "violet/airfrans-baseline-default" \
  --wandb-group "radford-airfrans-round1"
```

### What happened

This is the first AirfRANS production baseline on the radford branch. The run was capped at 6 epochs by the 30-minute timeout (SENPAI_TIMEOUT_MINUTES=30). The model was clearly still improving — val surface MSE dropped from 0.579 (epoch 1) to 0.397 (epoch 4) before bouncing back up at epochs 5-6 (likely cosine LR schedule oscillation with t_max=150 on a 6-epoch run). The loss was still declining at termination (1.63 → 1.31).

**Key observations:**
- **Pressure (p) dominates the error** — surface p MSE is 2.12 while Ux/Uy/nut are all <0.005. This is expected for AirfRANS and matches literature behavior.
- **Volume MSE is near 1.0** across all channels, meaning the model has barely begun learning volume predictions after 6 epochs. Surface predictions are ahead of volume, consistent with the surface refinement module kicking in.
- The cosine schedule with t_max=150 means the LR was still in its early phase — the model never reached the low-LR fine-tuning regime.
- **6 epochs is far too few for a meaningful baseline.** This needs 150 epochs (or at minimum 50+) to establish a proper baseline for comparison.

### Suggested follow-ups

1. **Run with full epoch budget** — the 30-min timeout severely limits this baseline. A longer run would show proper convergence behavior and give meaningful numbers for comparison.
2. **Add peak GPU memory logging** to train.py (e.g. `torch.cuda.max_memory_allocated()`) — this is missing and would be useful for all future experiments.
3. Once a proper baseline is established, investigate pressure-specific loss weighting to address the p-channel dominance.

**Comment by morganmcg1** [comment]
ADVISOR: This PR is supposed to establish the AirfRANS anchor, but one run is still under-using your node. Stay in the clean-default family and launch a compact stabilizer sweep in parallel.

Run at least:
- `v0`: exact baseline command from the PR body
- `v1`: same command but add `--no-use-lookahead`
- `v2`: same command but add `--no-use-ema`
- `v3`: same command but add both `--no-use-lookahead --no-use-ema`

Use one GPU per trial with `CUDA_VISIBLE_DEVICES`, keep the same W&B group, and suffix the run names. This will tell us whether the AirfRANS baseline should inherit the full Tandem-style optimizer scaffolding or whether the simpler training stack is better.

**Comment by morganmcg1** [comment]
ADVISOR: Clarification on the effective run budget for this `radford` sweep.

The assignment text may show `--epochs 150`, but the live senpai harness for this run is enforcing:
- `SENPAI_MAX_EPOCHS=50`
- `SENPAI_TIMEOUT_MINUTES=30`

So in practice, your run stops at 50 epochs or 30 minutes, whichever comes first. Treat this as the true budget when deciding how many parallel variants to launch and when interpreting results.

Please report for each variant:
- epochs actually completed
- whether it hit the 30-minute timeout or the 50-epoch cap
- the best validation metric reached within that enforced budget

Optimize for best performance under this actual capped regime, not for hypothetical longer training beyond the harness limits.

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2420 because Baseline established — thank you. Lion lr=3e-4 (0.538) is now superseded by AdamW lr=5e-4 (0.331, PR #2423, -38%). Your observation that 6 epochs is far too few is confirmed: the winner also ran only 6 epochs and was still improving. The epoch-budget problem gets addressed in Round 2 by reducing model_slices for faster throughput..
